frontend/src/scenes/data-warehouse/editor/multitabEditorLogic.tsx

```diff
@@ -93,18 +95,130 @@ const getStorageItem = async (key: string): Promise<string | null> => {
     }
 
     const lsValue = localStorage.getItem(key)
-
     if (lsValue) {
-        await set(key, lsValue)
-        localStorage.removeItem(key)
+        try {
+            await set(key, lsValue)
+            localStorage.removeItem(key)
+        } catch (error) {
+            // this can happen if the storage quota is exceeded, or if the user is in private mode when trying to migrate
+            posthog.captureException(new Error('sql-editor-local-storage-migration-failure'), { error })
+        }
         return lsValue
     }
-
     return null
 }
 
 const setStorageItem = async (key: string, value: string): Promise<void> => {
-    await set(key, value)
+    try {
+        await set(key, value)
+    } catch (error) {
+        // this would trigger if theres a problem writing to IndexedDB, and therefore we fall back to localStorage
+        posthog.captureException(new Error('sql-editor-indexeddb-write-failure'), { error })
+        localStorage.setItem(key, value)
+    }
+}
+
+const performComprehensiveMigration = async (key: string): Promise<QueryTab[] | null> => {
```

Comment:
My assumption here is that we just wanna move `allTabs` from localstorage into the indexedDB, right? This is quite a risky migration.